sources/osg/RenderStage.js

```diff
@@ -37,6 +30,41 @@ RenderStage.prototype = MACROUTILS.objectLibraryClass( MACROUTILS.objectInherit(
     // temporary, Utils.createPrototypeClass will solve this
     constructor: RenderStage,
 
+    cloneType: function () {
+        return this.constructor.prototype.getOrCreate().init();
```

Comment:
I dont think the cache needs to be low level, you can just let the cloneType create new class and have a cache by type in the cull visitor because it's here it's used.
like a CullVisitor.getOrCreateRenderStage( type, args )